src/StoreKeeper/WooCommerce/B2C/Exports/OrderExport.php

```diff
@@ -780,13 +842,25 @@ protected function processPaymentsAndRefunds(WC_Order $WpObject, int $storekeepe
                             $comment .= ' #'.$WpObject->get_transaction_id();
                         }
 
-                        $ShopModule->markOrderAsPaid(
-                            $storekeeper_id,
-                            [
-                                'comment' => $comment,
-                                'provider_method_type_id' => $provider_method_type_id,
-                            ]
-                        );
+                        $paymentId = $PaymentModule->newWebPayment([
+                            'amount' => $WpObject->get_total(),
+                            'description' => $comment,
+                        ]);
+
+                        if ($paymentId) {
+                            PaymentGateway::addPayment($WpObject->get_id(), $paymentId, $WpObject->get_total());
+                            // Check for the error being thrown.
+                            try {
+                                $shopModule->attachPaymentIdsToOrder(['payment_ids' => [$paymentId]], $storekeeper_id);
+                            } catch (GeneralException $generalException) {
+                                // Check if the payment is already attached to the order, we can ignore it an move on.
+                                $strpos = strpos($generalException->getMessage(), 'This payment is already linked');
```

Comment:
the full error is `This payment is already linked (id=$payment_id) to different order` this normally is bad :P.

Reply:
It was implemented like this before, I just moved it to a granular function :rofl: should we change this?